tutorials/tutorial_1/containers/prediction-model/Dockerfile

```diff
@@ -1,38 +1,64 @@
-# !! EDIT THIS: Set the version of Python to be used.
-FROM python:3.8-bullseye as wheelbuilder
+# !! EDIT THIS: Set base Docker image to be used.
+FROM nvcr.io/nvidia/cuda:12.6.3-runtime-ubuntu24.04
 
-COPY requirements.txt ./
-
-RUN --mount=type=cache,target=/root/.cache/pip \
-    python -m pip install -U pip setuptools wheel \
- && pip wheel --cache-dir=/.cache/pip --wheel-dir /wheels -r requirements.txt
+# Set env vars to be able to run apt-get commands without issues.
+ENV LC_ALL="C.UTF-8"
+ENV TZ=Etc/UTC
 
+# Install Python 3.12
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; \
+    echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    export DEBIAN_FRONTEND=noninteractive \
+ && apt-get update -qq \
+ && apt-get install -q -y --no-install-recommends -q python3 python3-dev python3-venv
 
+# Set up user and group.
+ARG UID=5642
+ARG GID=5642
+RUN <<"EOF" bash
+    set -eu -o pipefail
+    if [[ $UID -eq 0 ]]; then
+        # Symlink /home/localuser to the root user's home dir
+        home_dir="$(getent passwd $UID | cut -d: -f6)"
+        ln -s "$home_dir" /home/localuser
+    else
+        if [[ $UID -ge 1000 ]] && getent passwd $UID >/dev/null; then
```

Comment:
Why would there be an existing user? And if there is one - is it OK that we simply delete it?

Reply:
Good catch. Should probably remove this and the following if where it deletes the group as well right?

Reply:
Not sure - can you check with Tal? I think he wrote this code and I assume there was a reason, but not sure if we need it in our tutorial or not.

Reply:
On some linux systems, the uids for non-system users begin at 1000. So, for example, my user on my machine has UID 1000. However, sometimes UID 1000 already exists in the container image, such as with images based on Ubuntu 24.04. To run those with the same user as the local user, that hack is needed.

Reply:
The idea of running the container with the same UID and GID as the local user is to avoid permissions issues with files read and/or written by the container in directories mounted into it.

Reply:
So I guess whether to keep this depends on whether we want these containers to be compatible with `docker-run.sh` and `docker-run-nvflare.sh`, which try to run the containers using the user's UID and GID on the host machine.

Reply:
I imagine we want to keep it compatible, even if the tutorial no longer uses it. In any case, sounds like there's no real downside to keeping it in the code.